src/docfx/build/toc/ServicePageGenerator.cs

```diff
@@ -0,0 +1,114 @@
+            string directoryName = "")
+        {
+            if (node == null || string.IsNullOrEmpty(node.Name))
+            {
+                return;
+            }
+
+            var filename = Regex.Replace(node.Name, @"\s+", "");
+
+            if (!string.IsNullOrEmpty(node.LandingPageType.ToString()))
+            {
+                var baseDir = string.IsNullOrEmpty(_joinTOCConfig.OutputPath) ? Directory.GetCurrentDirectory() : _joinTOCConfig.OutputPath.Replace("..", "");
+                var servicePagePath = FilePath.Generated(new PathString($"./{baseDir}/{directoryName}/{filename}.yml"));
+
+                var name = node.Name;
+                var fullName = node.Name;
+                var uid = node.Uid;
+                var href = node.Href;
+
+                var children = new List<SourceInfo<ServicePageModel>>();
+                foreach (var item in node.Items)
+                {
+                    ServicePageModel newItem;
+                    var newItemName = item.Value.Name;
+                    var newItemHref = item.Value.Href;
+                    var newItemUid = item.Value.Uid;
+
+                    if (!string.IsNullOrEmpty(newItemHref.Value))
+                    {
+                        if (_repositoryProvider.Repository != null && item.Source != null)
+                        {
+                            var itemFilePath = Path.GetDirectoryName(Path.Combine(_repositoryProvider.Repository.Path, item.Source.File.Path));
+                            var itemHrefPath = Path.Combine(itemFilePath == null ? "" : itemFilePath, newItemHref.Value);
+                            var outDir = Path.Combine(_output.OutputPath, servicePagePath.Path);
+                            var newhref = Path.GetRelativePath(outDir == null ? "" : outDir, itemHrefPath);
+                            newItemHref = new SourceInfo<string?>(newhref);
+                        }
+
+                        newItem = new ServicePageModel(newItemName, newItemHref, new SourceInfo<string?>(null));
+                    }
+                    else
+                    {
+                        newItem = new ServicePageModel(newItemName, new SourceInfo<string?>(null), newItemUid);
+                    }
+
+                    children.Add(new SourceInfo<ServicePageModel>(newItem));
+                }
+
+                var langs = new List<string?>();
+                if (_joinTOCConfig.ContainerPageMetadata != null)
+                {
+                    foreach (var lang in _joinTOCConfig.ContainerPageMetadata)
+                    {
+                        if (lang.Key.Equals("langs") && lang.Value != null)
+                        {
+                            langs = lang.Value.ToObject<List<string?>>();
+                        }
+                    }
+                }
+                var landingPageType = node.LandingPageType;
+                var pageType = new SourceInfo<LandingPageType?>(landingPageType);
```

Comment:
If the source info cannot be populated the model does not need to declare fields as `SourceInfo<T>`. Presumably `new SourceInfo<T>` should never appear in this file, and if there are fields with source info to carry over, use `SourceInfo<T>.With` method.

Reply:
Fixed.